packages/ui/src/components/tabs.tsx

```diff
@@ -116,8 +138,8 @@ export function Tabs({
           </Primitive.TabsTrigger>
         ))}
       </Primitive.TabsList>
-      <ValueChangeContext.Provider value={onValueChange}>
-        {props.children}
+      <ValueChangeContext.Provider value={{ onValueChange, valueToIdMap: valueToIdMapRef.current }}>
```

Comment:
Add useMemo for context value to avoid unnecessary rerenderings

Reply:
added `useMemo` in f52531a - thanks!